dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,233 @@
+import React, { useState, useEffect } from "react";
+import {useDispatch} from 'react-redux'
+import "./index.css";
+import {
+  ToggleGroup,
+  ToggleGroupItem,
+  Page,
+  Masthead,
+  MastheadToggle,
+  PageSidebar,
+  PageToggleButton,
+  PageSection,
+  PageSectionVariants,
+} from "@patternfly/react-core";
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+} from "@patternfly/react-table";
+import BarsIcon from "@patternfly/react-icons/dist/js/icons/bars-icon";
+import SearchBox from "../SearchComponent";
+import DatePickerWidget from "../DatePickerComponent";
+import Heading from "../HeadingComponent";
+import PathBreadCrumb from "../BreadCrumbComponent";
+import AlertMessage from "../AlertComponent";
+import NavItems from "../NavbarComponent";
+import EmptyTable from "../EmptyStateComponent";
+import moment from "moment";
+import { fetchPublicDatasets } from "../../../actions/fetchPublicDatasets";
+import TablePagination from "../PaginationComponent";
+let startDate = moment(new Date(1990, 10, 4)).format("YYYY-MM-DD");
+let endDate = moment(new Date()).format("YYYY-MM-DD");
+let controllerName = "";
+let dataArray = [];
+export const TableWithFavorite = () => {
+  const columnNames = {
+    controller: "Controller",
+    name: "Name",
+    creationDate: "Created On",
+  };
+  const [activeSortIndex, setActiveSortIndex] = useState(null);
+  const [activeSortDirection, setActiveSortDirection] = useState(null);
+  const [favoriteRepoNames, setFavoriteRepoNames] = useState([]);
+  const [publicData, setPublicData] = useState([]);
+  const [isSelected, setIsSelected] = useState("controllerListButton");
+  const [isNavOpen, setIsNavOpen] = useState(false);
+  const [page,setPage]=useState(1);
+  const [perPage,setPerPage]=useState(10);
+  const dispatch=useDispatch();
+  useEffect(() => {
+     dispatch(fetchPublicDatasets()).then((res) => {
+        dataArray = res.data;
+        setPublicData(res.data);
+        setFavoriteRepoNames(JSON.parse(localStorage.getItem("favControllers")))
+      })
+      .catch((err) => {
+        console.log(err);
+      });
```

Comment:
I think we probably need a consistent framework for handling unexpected errors, rather than just throwing around clauses like this to log on the console (which most users will never see). We need to think about what this means and how the dashboard can reliably recover and continue ... or, perhaps, in the worst case, throw up a panic dialog box and reload the App from scratch. Seems there must be some "best practice" and "tradition" in this area.

(I don't mean that necessarily has to happen in this PR, but at least we should construct some story in Jira, perhaps under the "robust dashboard" hierarchy.)

Reply:
Yes,we can start working in that direction.
I also agree that the framework for handling errors should be consistent,and by consistent I mean we should follow the same approach while dealing with all unexpected errors,what method or framework we should follow is a point of discussion.